types/ldapjs-client/index.d.ts

```diff
@@ -0,0 +1,92 @@
+// Type definitions for ldapjs-client 0.1
+// Project: https://github.com/zont/ldapjs-client#readme
+// Definitions by: Valerio Coltrè <https://github.com/colthreepv>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.0
+
+export class Filter {
+    matches(obj: any): boolean;
+    type: string;
+}
+
+export interface Change {
+    operation: string;
+    modification: {
+        [key: string]: any;
+    };
+}
+
+export interface SearchOptions {
+    scope: string;
+    filter: string | Filter;
+    attributes: string[];
+    sizeLimit: number;
+    timeLimit: number;
+    typesOnly: boolean;
+}
+export type PartialSearchOptions = Partial<SearchOptions>;
+
+export interface ClientOptions {
+    url: string;
+    tlsOptions?: object;
+    timeout?: number;
+}
+
+export class LdapClient {
+    constructor(options: ClientOptions);
+
+    /**
+     * Adds an entry to the LDAP server.
+     */
+    add(dn: string, entry: object): Promise<any>;
+
+    /**
+     * Performs a simple authentication against the server.
+     */
+    bind(dn: string, password: string): Promise<any>;
+
+    /**
+     * Deletes an entry from the LDAP server.
+     */
+    del(dn: string): Promise<any>;
+
+    /**
+     * Disconnect from the LDAP server and do not allow reconnection.
+     *
+     * If the client is instantiated with proper reconnection options, it's
+     * possible to initiate new requests after a call to unbind since the client
+     * will attempt to reconnect in order to fulfill the request.
+     *
+     * Calling destroy will prevent any further reconnection from occurring.
+     */
+    destroy(): Promise<void>;
+
+    /**
+     * Performs an LDAP modify against the server.
+     *
+     * @param dn the DN of the entry to modify.
+     * @param change update to perform (can be [Change]).
+     */
+    modify(dn: string, change: Change): Promise<any>;
+
+    /**
+     * Performs an LDAP modifyDN against the server.
+     */
+    modifyDN(dn: string, newName: string): Promise<any>;
+
+    /**
+     * Performs an LDAP search against the server.
+     *
+     * Note that the defaults for options are a 'base' search.
+     */
+    // tslint:disable-next-line: no-unnecessary-generics
+    search(base: string, options: PartialSearchOptions): Promise<unknown[]>;
+
+    /**
+     * Unbinds this client from the LDAP server.
+     *
+     */
+    unbind(): Promise<any>;
+}
+
+export default LdapClient;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
Ok perfect. Sorry for have missed that part.

There is a problem though, I don't seem to be able to export interfaces and types alongside the default export being LdapClient

Is there a way in Typescript to do this? I have googled but not many people use commonjs exports  in the wild (outside definitelytyped)